### PR TITLE
Use aws-credentials action to configure creds bc it uses OIDC by default

### PR DESCRIPTION
**Pull request by NathanielRN**
*Issue #, if available:*

N/A

*Description of changes:*

GitHub just announced [their OIDC feature as GA](https://github.blog/changelog/2021-10-27-github-actions-secure-cloud-deployments-with-openid-connect/), but what’s even cooler is that they seem to have worked with the “configure-aws-credentials GH Action” team to integrate it seamlessly into the action :]

From the [aws-actions/configure-aws-credentials updated README](https://github.com/aws-actions/configure-aws-credentials#assuming-a-role) from 15 days ago:

> We recommend using GitHub’s OIDC provider to get short-lived credentials needed for your actions. Specifying role-to-assume without providing an aws-access-key-id  or a web-identity-token-file will signal to the action that you wish to use the OIDC provider. The default session duration is 1 hour when using the OIDC provider to directly assume an IAM Role. The default session duration is 6 hours when using an IAM User to assume an IAM Role (by providing an aws-access-key-id, aws-secret-access-key, and a role-to-assume) . If you would like to adjust this you can pass a duration to role-duration-seconds, but the duration cannot exceed the maximum that was defined when the IAM Role was created. The default session name is GitHubActions, and you can modify it by specifying the desired name in role-session-name.

This should mean we can use the action to do OIDC authentication, and that the Soak Tests can also use it because the 1 hour default expiry will be enough for it to refresh those credentials 🙂 Previously this was not possible because the OIDC token had a default expiry of 5 minutes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/soak-testing.yml
Comment (willarmiros) [inline .github/workflows/soak-testing.yml:103]: So these were for a user due to the previous OIDC limitations, but now we can remove the user and these long term creds right?

**Review by willarmiros — APPROVED** [review]
Wow, it's like they read our wishlist!

Just to confirm my understanding - we will be using the new default of the configure-aws-credentials which uses the GitHub OIDC auth under the hood. This will mean 1 hour assumed role duration for all actions (which is more than sufficient) except the soak tests, which we configure to 6 hours. Correct?

Also will want to wait for @anuraaga's approval here.

File: .github/workflows/soak-testing.yml
Comment (NathanielRN) [reply .github/workflows/soak-testing.yml:103]: Use it was for a user, correct!

Well originally we could not use OIDC because [the token only lasted for 5 minutes](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/90). OIDC is able to refresh the sessional credentials on every AWS call, but the Soak Tests might wait up to 10 minutes to make another AWS call.

The hope is that now, with a 1 hour expiry, that is plenty of time for the Soak Tests to refresh credentials every time it polls the alarms (every 10 minutes) to see if a spike occurred. If the Soak Tests ever poll longer than that then we would have a bigger problem...

However, I have not tested this on my soak-testing-demo fork and was counting on just merging it and fixing it if it breaks 😅... I can move this file to a separate PR if we want to test it on just the Integration Tests first 🙂

**Review by anuraaga — APPROVED** [review]
Cool, let's try this!

**Comment by anuraaga** [comment]
Perhaps the new code hasn't been released yet?

https://github.com/aws-observability/aws-otel-java-instrumentation/runs/4099357345?check_suite_focus=true

Let me revert this for now since I'll probably be releasing a patch today and need main build up

**Comment by willarmiros** [comment]
> Perhaps the new code hasn't been released yet?
> 
> https://github.com/aws-observability/aws-otel-java-instrumentation/runs/4099357345?check_suite_focus=true
> 
> Let me revert this for now since I'll probably be releasing a patch today and need main build up

Lol yep https://github.com/aws-actions/configure-aws-credentials/issues/302

**Pull request merged**
